src/Unused/Types.hs

```diff
@@ -2,17 +2,49 @@ module Unused.Types
     ( TermMatch(..)
     , TermMatchSet
     , ParseResponse(..)
+    , responseFromParse
+    , withOneFile
+    , withOneOccurrence
+    , skipRailsExtras
     ) where
 
 import Text.Parsec (ParseError)
-import Data.Map.Strict as Map
+import qualified Data.Map.Strict as Map
+import Data.List (isInfixOf)
+import Unused.Util (groupBy)
 
 data TermMatch = TermMatch
     { term :: String
     , path :: String
     , occurrences :: Int
     } deriving Show
 
-type TermMatchSet = Map String [TermMatch]
+type TermMatchSet = Map.Map String [TermMatch]
+
 
 data ParseResponse = ValidParse TermMatchSet | InvalidParse ParseError
```

Comment:
Oooh, I'd gone down the functor route in https://github.com/joshuaclayton/unused/pull/1/commits/966806ad334d069372451b2d42768356caba7be2 but this approach is appealing.

I'd avoided that parser because it seemed like too much of an abstraction - specifically, it wouldn't use any actual Parsec under the covers, instead transforming the previously parsed data. If that's normal (I don't know!) I'd be fine making the switch.